pkgs/by-name/sv/svchttpx/package.nix

```diff
@@ -0,0 +1,37 @@
+{ lib, fetchFromGitHub, graalvm17-ce, maven }:
+let
+  mavenJdk17 = maven.override {
+    jdk = graalvm17-ce;
+  };
+in
+mavenJdk17.buildMavenPackage rec {
+  pname = "svchttpx";
+  version = "8e4e52708d96282b42aed6f0381000476a5486d4";
+
+  src = fetchFromGitHub {
+    owner = "ghostbuster91";
```

Comment:
for now I had to set it to my fork in order to make it work. 
Here are the changes: https://github.com/servicex-sh/httpx/compare/master...ghostbuster91:httpx:master

so basically I downloaded graalvm-reachability-metadata and modified pom to use it instead of downloading a particular version from the web. 

Now, I think that it is rather inconvenient to do it like that and I would like to propose such changes to upstream httpx package.

Reply:
Why do you have to do this it's not clear?

You can also patch the source via Nix without pulling from a fork which is pretty common in Nix since a lot of software needs to be modified to work in nixpkgs.

Reply:
> Why do you have to do this it's not clear?

Not sure how much you know about graalvm native image. TLDR it compiles your java application into an executable binary. The binary is compiled together with a small jvm (substrate-vm) that is stripped from a lot of things like e.g. JIT. Code optimization happens AOT. 

A lot of java libraries rely on reflection that is a purely runtime mechanism. In order to producer final binary the compiler needs to resolve all reflection calls during the compile time. This is a tricky task and a lot of different techniques are used to get the best results. 

Now, I am not an expert on this, but my understanding is that the metadata in our case is just combined analysis of reflection calls of popular libraries in jvm ecosystem like e.g. gson. 

So without that data the reflection substitution might not be fully correct which is exactly the case in here, as if you remove the metadata definition from the pom.xml the build will fail. 

> You can also patch the source via Nix without pulling from a fork which is pretty common in Nix since a lot of software needs to be modified to work in nixpkgs.

:) I was afraid that you would say that. I guess this is what I will do. Should I use `sed` to modify `pom.xml` or do you know any better alternative? I was looking at `xq` but it does not seem to support modifications.